lib/rubygems/gemcutter_utilities.rb

```diff
@@ -258,15 +260,42 @@ def get_scope_params(scope)
     else
       say "Please select scopes you want to enable for the API key (y/n)"
       API_SCOPES.each do |scope|
-        selected = ask "#{scope} [y/N]: "
-        scope_params[scope] = true if selected =~ /^[yY](es)?$/
+        selected = ask_yes_no("#{scope}", false)
```

Comment:
Is there any particular reason to use `false` as a default? If I understand it well on non-TTY environments it will fail anyway (since the other `ask_yes_no`). We can provide on default and update the original message to highly recommend to reduce the scope of the key to the minimum according to the API key purpose. Maybe we can also list all scopes in this message to make it clear which scopes are you going to be asked like `Please select scopes (scope1, scope2,)` and I think where is no need for the `(y/n)` part in the message.

Reply:
@simi Providing a default makes it so the user can simply hit "enter" instead of actually choosing yes or no. Otherwise, if you don't provide a response it will ask you the question again. Since there are so many options, I find I can get through the sign on process much more quickly with having a default. (one key press for enter instead of two, one for `n` and then enter)

> Maybe we can also list all scopes in this message to make it clear which scopes are you going to be asked like Please select scopes (scope1, scope2,)
There are seven scopes we ask about, so I feel it may be a bit overwhelming to list them all at once.

Either way, the scopes section here is not a new addition, this is the current behavior. So if we'd like to change it significantly I would suggest that be in a separate PR to give it the attention it deserves

Reply:
OK, seems reasonable.